### PR TITLE
update MediaTypeNames

### LINKED ISSUE #1489: System.Net.Mime.MediaTypeNames missing types
https://github.com/dotnet/corefx/blob/3f0bd5d8c3d019ad8b1561395030a58bc38ddfe9/src/System.Net.Mail/ref/System.Net.Mime.cs#L44

I tried using `System.Net.Mime.MediaTypeNames` in my asp net core api to remove "magic strings"

```c#
[Consumes(System.Net.Mime.MediaTypeNames.Application...)]
public IActionResult Post(...)
{ ... }
```

But it's missing quite a few MIME types.

These come to mind:

> text/css
> text/javascript
> 
> application/json
> application/json-patch+json
> application/xml
> application/javascript
> application/x-www-form-urlencoded
> 
> multipart/form-data
> multipart/byteranges
> 
> image/png
> image/bmp
> image/webp
> image/svg+xml
> 
> audio/midi
> audio/mpeg
> audio/webm
> audio/ogg
> audio/wav
> 
> video/webm
> video/ogg

### LINKED ISSUE #85807: [API Proposal]: add more entries to MediaTypeNames
### Background and motivation

This is primarily community driven, long discussion in #1489. This is primarily convenience to somewhat match common patterns for web applications. 

Additional sources are: [IANA Media Types](https://www.iana.org/assignments/media-types/media-types.xhtml)

### API Proposal

```diff
namespace System.Net.Mime;

    public static class MediaTypeNames
    {
        public static class Text
        {
            public const string Plain = "text/plain";
            public const string Html = "text/html";
            public const string Xml = "text/xml";
            public const string RichText = "text/richtext";
+           public const string Css = "text/css";                 // RFC2318
+           public const string Csv = "text/csv";                 // RFC4180
+           public const string JavaScript = "text/javascript";   // RFC9239
+           public const string Markdown = "text/markdown";       // RFC7763
+           public const string Rtf = "text/rtf";
        }

        public static class Application
        {
            public const string Soap = "application/soap+xml";
            public const string Octet = "application/octet-stream";
            public const string Rtf = "application/rtf";
            public const string Pdf = "application/pdf";
            public const string Zip = "application/zip";
            public const string Json = "application/json";
            public const string Xml = "application/xml";
+           public const string FormUrlEncoded = "application/x-www-form-urlencoded"; 
+           public const string JsonPatch = "application/json-patch+json"; // RFC6902
+           public const string JsonSequence = "application/json-seq";     // RFC7464
+           public const string Manifest = "application/manifest+json";    // W3C
+           public const string ProblemJson = "application/problem+json";  // RFC7807
+           public const string ProblemXml = "application/problem+xml";    // RFC7807
+           public const string Wasm = "application/wasm";                 // W3C
+           public const string XmlDtd = "application/xml-dtd";            // RFC7303
+           public const string XmlPatch = "application/xml-patch+xml";    // RFC735
        }

        public static class Image
        {
            public const string Gif = "image/gif";
            public const string Tiff = "image/tiff";
            public const string Jpeg = "image/jpeg";
+           public const string Avif = "image/avif";
+           public const string Bmp = "image/bmp";     // RFC7903
+           public const string Icon = "image/x-icon"; 
+           public const string Png = "image/png";     // W3C
+           public const string Svg = "image/svg+xml"; // W3C
+           public const string Webp = "image/webp";
        }

+        public static class Multipart
+        {
+           public const string ByteRanges = "multipart/byteranges";  // RFC9110
+           public const string FormData = "multipart/form-data";     // RFC7578
+        }

+        public static class Font
+        {
+           public const string Collection = "font/collection"; // RFC8081
+           public const string Otf = "font/otf";               // RFC8081
+           public const string Sfnt = "font/sfnt";             // RFC8081
+           public const string Ttf = "font/ttf";               // RFC8081
+           public const string Woff = "font/woff";             // RFC8081
+           public const string Woff2 = "font/woff2";           // RFC8081
+        } 
    }
```

### API Usage

there is not much, this is just convenience constant. 

### Alternative Designs

_No response_

### Risks

small. Added constants will increase assembly size a tiny little bit.

### PR DESCRIPTION
**Pull request by wfurt**
fixes #85807
fixes https://github.com/dotnet/runtime/issues/1489

ref assembly change seems large -> result of running `dotnet msbuild /t:GenerateReferenceAssemblySource`.
I could probably just update it by hand to minimize diff but it seems better to sync up with current scripts.

### CODE REVIEW

File: src/libraries/System.Net.Mail/ref/System.Net.Mail.cs [hunk 1]
```diff
@@ -8,12 +8,12 @@ namespace System.Net.Mail
 {
     public partial class AlternateView : System.Net.Mail.AttachmentBase
     {
-        public AlternateView(System.IO.Stream contentStream) : base (default(System.IO.Stream)) { }
-        public AlternateView(System.IO.Stream contentStream, System.Net.Mime.ContentType? contentType) : base (default(System.IO.Stream)) { }
-        public AlternateView(System.IO.Stream contentStream, string? mediaType) : base (default(System.IO.Stream)) { }
-        public AlternateView(string fileName) : base (default(System.IO.Stream)) { }
-        public AlternateView(string fileName, System.Net.Mime.ContentType? contentType) : base (default(System.IO.Stream)) { }
-        public AlternateView(string fileName, string? mediaType) : base (default(System.IO.Stream)) { }
+        public AlternateView(System.IO.Stream contentStream) : base (default(string)) { }
```
Comment (ManickaP) [inline src/libraries/System.Net.Mail/ref/System.Net.Mail.cs:11]: I don't know if it matters here, but this calls to a different base constructor than it used to and than what the actual source code calls. Is that OK (i.e. non-breaking)?

**Review by ManickaP — APPROVED** [review]
LGTM modulo that one comment. Thanks!

File: src/libraries/System.Net.Mail/ref/System.Net.Mail.cs (hunk 1 above)
Comment (jkotas) [reply src/libraries/System.Net.Mail/ref/System.Net.Mail.cs:11]: > I don't know if the ref assembly matters as long as it shows correct constructor prototypes

It does not matter. The base constructor call is there just to make C# compiler happy.

It looks like a non-determinism in the auto-generator. It used to call `string` overload, then it flipped to Stream overload in https://github.com/dotnet/corefx/pull/35557/, and it is flipping back to string overload again.

**Review by jkotas — APPROVED** [review]

File: src/libraries/System.Net.Mail/ref/System.Net.Mail.cs (hunk 1 above)
Comment (ericstj) [reply src/libraries/System.Net.Mail/ref/System.Net.Mail.cs:11]: As @jkotas mentioned the base constructor call is only required to make it compile.  The one chosen is based on a heuristic: https://github.com/dotnet/arcade/blob/e375d9151b2345e26a0bf967913fa4d621891496/src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs#L429

Typically those are stable based on metadata order, or they are ordinal sorted based on the signature.  I could imagine a problem where GenAPI might have been missing references, and sorted `System.IO.Stream` before `System.String` - that could have happened when https://github.com/dotnet/corefx/pull/35557 was submitted.

If you notice this changing a lot on rerun of the tool in the current codebase, that'd be a bug to fix.  So long as it's stable now I'd just ignore this change and accept it to align with the current output.

**Pull request merged**